### ISSUE TITLE
Questions regarding the train and testing splitting

### ISSUE THREAD

**Issue by dhimmel** [OpenBioLink/OpenBioLink#7]
The partitioning of train and testing edges is a challenging topic. Quoting from the paper:

> With the aid of the train-test-split module, the data can be divided into a training and a test set, either randomly or via time slices. Special focus was put on the robustness and difficulty of the test set, which contains only entities that are also present in the training set and does not contain relations that can be trivially inferred from the training set (such as reverse edges of symmetric relations, inverse relations or super-relations). Negative samples are produced using the negative edges present in the benchmark data sources and - where needed - by applying typed negative sampling.

Quoting from the readme:

```
   --mode rand|time    Mode of train-test-set split, options=[rand, time], (default=rand)
```

Cool to see that there is an option to split at by time. Recent work such as [this paper](https://doi.org/10.1186/s12859-019-3297-0 "Time-resolved evaluation of compound repositioning predictions on a text-mined knowledge network") has shown that predicting the future is harder than predicting parts of the past that are withheld. What edge types in the the OpenBioLink network have date information?

Our research has found that network degree alone is [able to predict heldout edges](https://greenelab.github.io/xswap-manuscript/v/3d636da3a8f52a4d44c78cdf600f1fa9fbb1dbd9/#fig:discrimination) to a large extent, oftentimes with AUROC > 85% for many biomedical edge types. Did you investigate how degrees vary between "typed negative sampling" and positive?

I suspect that while the test set "does not contain relations that can be trivially inferred from the training set", it would be possible to achieve very high performance using node degree information only.

**Comment by matthias-samwald** [comment]
Currently only some of the datasets allow for timeslicing. @annabreit please add information to readme which.

**Comment by matthias-samwald** [comment]
Will look into prediction based on node degree and XSwap!

**Comment by cthoyt** [comment]
Bernoulli negative sampling is one solution to dealing with node degrees during negative sampling (as compared to uniform negative sampling). It was proposed by Wang *et al.* in [Knowledge Graph Embedding by Translating on Hyperplanes](https://www.aaai.org/ocs/index.php/AAAI/AAAI14/paper/view/8531).

**Comment by matthias-samwald** [comment]
→ We have added information on which data sources currently allow for time-sliced benchmark creation to the table listing data sources in the README. We will investigate the effect of degree-preserving graph permutation (such as those produced by XSwap) and alternative sampling technques such as Bernoulli negative sampling in follow-up work.
